checker/src/org/checkerframework/checker/guieffect/GuiEffectVisitor.java

```diff
@@ -158,16 +227,75 @@ public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
             // Instantiate type-polymorphic effects
             if (srcType.hasAnnotation(AlwaysSafe.class)) {
                 targetEffect = new Effect(SafeEffect.class);
+                polyUIArgumentsAreUI = false;
             } else if (srcType.hasAnnotation(UI.class)) {
                 targetEffect = new Effect(UIEffect.class);
             }
             // Poly substitution would be a noop.
         }
 
-        Effect callerEffect = atypeFactory.getDeclaredEffect(callerElt);
-        // Field initializers inside anonymous inner classes show up with a null current-method ---
-        // the traversal goes straight from the class to the initializer.
-        assert (currentMethods.peek() == null || callerEffect.equals(effStack.peek()));
+        // Look for lambda arguments and infer them to be @UIEffect for this call
+        List<? extends ExpressionTree> args = node.getArguments();
+        List<? extends VariableElement> parameters = methodElt.getParameters();
+        Pair<AnnotatedExecutableType, List<AnnotatedTypeMirror>> mfuPair =
+                atypeFactory.methodFromUse(node);
+        AnnotatedExecutableType invokedMethod = mfuPair.first;
+        List<AnnotatedTypeMirror> typeargs =
+                AnnotatedTypes.expandVarArgs(atypeFactory, invokedMethod, node.getArguments());
+        for (int i = 0; i < args.size(); ++i) {
+            if (args.get(i) instanceof LambdaExpressionTree) {
+                //System.err.println(methodElt + " is from stub file? " + atypeFactory.isFromStubFile(methodElt));
+                LambdaExpressionTree lambdaTree = (LambdaExpressionTree) args.get(i);
+                VariableElement formalArgElement = parameters.get(i);
+                // For some reason, these two have different information, and neither is a subset of the other.
+                // argTypeFromMFU is needed for stubs, argTypeFromArgElement will actually find @PolyUI when in the
+                // source. Neither will find @PolyUI in stubs.
+                // ToDo: Figure out a better way to works in both cases
+                AnnotatedTypeMirror argTypeFromArgElement =
+                        atypeFactory.getAnnotatedType(formalArgElement);
+                AnnotatedTypeMirror argTypeFromMFU = typeargs.get(i);
```

Comment:
@mernst @wmdietl Is this discrepancy @lazaroclapp observed normal?  Is there a single way to retrieve the annotations from either the source or stub, or is this a quirk/bug of the CF w.r.t. lambdas?